Source/JavaScriptCore/dfg/DFGLoopUnrollingPhase.cpp

```diff
@@ -0,0 +1,822 @@
+            ++count;
+        });
+        for (size_t i = count; i < expectedCount; ++i)
+            m_graph.m_varArgChildren.append(Edge());
+
+        Node* clone = cloneAndAppend(node);
+        clone->children.setFirstChild(firstChild);
+        return clone;
+    }
+    case Branch: {
+        Node* clone = cloneAndAppend(node);
+        clone->setOpInfo(OpInfo(m_graph.m_branchData.add(WTFMove(*node->branchData()))));
+        clone->child1() = cloneEdge(node->child1());
+        return clone;
+    }
+    case CheckVarargs: {
+        if (node->hasVarArgs()) {
+            size_t firstChild = m_graph.m_varArgChildren.size();
+            size_t count = 0;
+            m_graph.doToChildren(node, [&](Edge& edge) {
+                m_graph.m_varArgChildren.append(cloneEdge(edge));
+                ++count;
+            });
+
+            Node* clone = cloneAndAppend(node);
+            clone->children.setFirstChild(firstChild);
+            return clone;
+        }
+        FALLTHROUGH;
+    }
+    case ExitOK:
+    case LoopHint:
+    case GetButterfly:
+    case JSConstant:
+    case Jump:
+    case CompareLess:
+    case CompareEq:
+    case CheckStructure:
+    case ArithBitNot:
+    case ArrayifyToStructure:
+    case ArithBitAnd:
+    case ArithBitOr:
+    case ArithBitRShift:
+    case ArithBitLShift:
+    case ArithBitXor:
+    case BitURShift:
+    case ArithAdd:
+    case ArithSub:
+    case ArithMul:
+    case ArithDiv:
+    case ArithMod:
+    case CheckArray:
+    case FilterCallLinkStatus:
+    case GetLocal:
+    case MovHint:
+    case Flush:
+    case ZombieHint:
+    case SetLocal:
+    case PhantomLocal:
+    case Check:
+    case AssertNotEmpty:
+    case SetArgumentDefinitely:
+    case NewArrayWithSize:
+    case NewArrayWithConstantSize:
+    case ValueToInt32:
+    case InvalidationPoint: {
+        Node* clone = cloneAndAppend(node);
+        clone->child1() = cloneEdge(node->child1());
+        clone->child2() = cloneEdge(node->child2());
+        clone->child3() = cloneEdge(node->child3());
+        return clone;
+    }
+    default:
+        dataLogLnIf(verbose, "Could not clone node: ", node, " into ", into);
+        RELEASE_ASSERT_NOT_REACHED();
+        return nullptr;
+    }
+}
+
+// FIXME: Add more condition and update operations if they are profitable.
+bool LoopUnrollingPhase::isSupportedConditionOp(NodeType op)
+{
+    switch (op) {
+    case CompareLess:
+    case CompareLessEq:
+    case CompareGreater:
+    case CompareGreaterEq:
+    case CompareEq:
+        return true;
+    default:
+        return false;
+    }
+}
+
+bool LoopUnrollingPhase::isSupportedUpdateOp(NodeType op)
+{
+    switch (op) {
+    case ArithAdd:
+    case ArithSub:
+    case ArithMul:
+    case ArithDiv:
+        return true;
+    default:
+        return false;
+    }
+}
+
+std::function<bool(CheckedInt32, CheckedInt32)> LoopUnrollingPhase::comparisonFunction(Node* condition)
```

Comment:
Let's just return a function pointer as you can cast non-capturing lambda to a function pointer, like,
```
bool(*)(CheckedInt32, CheckedInt32)
```